openedx/core/djangoapps/notifications/tasks.py

```diff
@@ -133,11 +139,12 @@ def send_notifications(user_ids, course_key: str, app_name, notification_type, c
     grouping_function = NotificationRegistry.get_grouper(notification_type)
     waffle_flag_enabled = ENABLE_NOTIFICATION_GROUPING.is_enabled(course_key)
     grouping_enabled = waffle_flag_enabled and group_by_id and grouping_function is not None
-    notifications_generated = False
-    notification_content = ''
+    generated_notification = None
     sender_id = context.pop('sender_id', None)
     default_web_config = get_default_values_of_preference(app_name, notification_type).get('web', False)
     generated_notification_audience = []
+    push_notification_audience = []
+    is_push_notification_enabled = ENABLE_PUSH_NOTIFICATIONS.is_enabled()
```

Comment:
It is better to use `CourseWaffleFlag` here. This will give more control when testing on stage and on prod